staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go

```diff
@@ -96,33 +96,39 @@ func BuildSwagger(crd *apiextensions.CustomResourceDefinition, version string) (
 
 	routes := make([]*restful.RouteBuilder, 0)
 	root := fmt.Sprintf("/apis/%s/%s/%s", b.group, b.version, b.plural)
+
+	var params []*restful.Parameter
+	nameParam := b.ws.PathParameter("name", "name of the "+b.kind).DataType("string")
+	namespaceParam := b.ws.PathParameter("namespace", "object name and auth scope, such as for teams and projects").DataType("string")
+
 	if b.namespaced {
-		routes = append(routes, b.buildRoute(root, "", "GET", "list", sampleList).
+		params = append(params, namespaceParam)
+		routes = append(routes, b.buildRoute(root, "", "GET", "list", sampleList, params).
 			Operation("list"+b.kind+"ForAllNamespaces"))
 		root = fmt.Sprintf("/apis/%s/%s/namespaces/{namespace}/%s", b.group, b.version, b.plural)
 	}
-	routes = append(routes, b.buildRoute(root, "", "GET", "list", sampleList))
-	routes = append(routes, b.buildRoute(root, "", "POST", "create", sample).Reads(sample))
-	routes = append(routes, b.buildRoute(root, "", "DELETE", "deletecollection", status))
+	routes = append(routes, b.buildRoute(root, "", "GET", "list", sampleList, params))
+	routes = append(routes, b.buildRoute(root, "", "POST", "create", sample, params).Reads(sample))
+	routes = append(routes, b.buildRoute(root, "", "DELETE", "deletecollection", status, params))
 
```

Comment:
at this point we could simply add `nameParam` to `params` and pass `params` to the `buildRoute` method for all invocations bellow. WDYT ?

Reply:
good point